### PR TITLE
Sam/gltf loader

### PR DESCRIPTION
**Pull request by samjgorman**
### Description of Changes
Introduce a lightweight abstraction for loading gltf objects & scaling and positioning these objects within a scene.

Screen Shot 2022-03-15 at 23 20 49

### Link to Notion ticket
https://www.notion.so/irisapp/Add-in-an-object-to-the-scene-that-comes-from-an-external-source-509f77ae4adf481abb25b762d9565597
https://www.notion.so/irisapp/Transfer-gltf-asset-hosted-locally-to-S3-in-sam-gltf-loader-make-clear-how-teammates-can-access-S3-72ad13699c5e4051b2798105059cae61

### Testing Procedure
Tested locally in the browser on valid local & s3 gltf paths.

### CODE REVIEW

**Comment by samjgorman** [comment]
@kanyesthaker  ready for review! 

One edge case we can consider as well is how to handle cases in which we provide an invalid path.  This will currently break the application.  We can either decide that is a situation that will be avoided and move on, or add some error handling to catch this error.

File: src/Player/Player.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import React, { useEffect, useRef } from 'react';
 import { playerMovementControls } from './playerMovementControls';
 import playerMovementEmitter from './playerMovementEmitter';
 import { PlayerPosition, PlayerVelocity } from '../allTypes';
-import { roundEntriesInVector, round } from './playerMovementHelpers';
+import { roundEntriesInVector, round } from './playerMovementUtils';
```
Comment (kanyesthaker) [inline src/Player/Player.tsx:7]: appreciate the name change for clarity!

File: src/Scene/GLTFModel.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * modelPath -> Path to gltf or glb folder or file. Paths can either be local or reference an S3 bucket.
+ * scaleFactor -> Factor to scale the x,y,z of the gltf.scene object.
+ * position -> Array of x,y,z coordinates to define the primitive's position in the scene. This is a small abstraction for
+ *  the native position property present in scene objects, as primitive elements require their position to be directly modified. 
+ *
+ * @returns model to render
+ */
+function GLTFModel(props: ModelProps) {
+  const gltf = useLoader(GLTFLoader, props.modelPath);
+  gltf.scene.scale.multiplyScalar(props.scaleFactor);
```
Comment (kanyesthaker) [inline src/Scene/GLTFModel.tsx:23]: nice, love how you cleaned this up

**Review by kanyesthaker — APPROVED** [review]
lgtm! clean, straightforward, and modular code -- nice work! let's not spend time on edge case error handling for now (since we're the only ones uploading these URLs) until it becomes a substantial problem

**Pull request merged**